src/rf/js/src/home/components/library.js

```diff
@@ -356,7 +356,8 @@ var LayerItem = React.createBackboneClass({
     },
 
     deleteLayer: function() {
-        console.log('delete layer');
+        this.getModel().destroy();
+        settings.getPendingLayers().remove(this.getModel());
```

Comment:
It seems strange to me that you have to do this. I would have thought that since the models have the same id, they'd be treated as the same and removing one would remove it from all the collections. I guess since they have different endpoints, they get different backbone ids but in any case it feels like we are taking over responsibility for managing data in two places when controlling a single model.